dothething/dothething/ClipsView.swift

```diff
@@ -12,20 +12,19 @@ struct ClipView: View {
     let clip: Clip
 
     var body: some View {
-        // thumbnail of video
-        Image(uiImage: clip.thumbnail)
-            .resizable()
-            .frame(width: 108, height: 192)
-            .cornerRadius(8)
-            .onTapGesture {
-                print("Tapped \(clip.url)")
-                Thinger.playVideo(videoUrl: clip.url)
+        GeometryReader { geometry in
+            if clip.thumbnail == UIImage() {
+                Image("Placeholder")
+                    .resizable()
+            } else {
+                Image(uiImage: clip.thumbnail)
+                    .resizable()
+                    .onTapGesture {
+                        print("Tapped \(clip.url)")
+                        Thinger.playVideo(videoUrl: clip.url)
+                    }
```

Comment:
The changes in this section replace the previous thumbnail image display with a more robust implementation. Now, if the thumbnail image is not available (i.e., `clip.thumbnail == UIImage()`), the code will display the "Placeholder" image instead. This is a good way to handle missing or unavailable thumbnail images, ensuring a consistent user experience.